testsuite/tests/mgc/test_basic.py

```diff
@@ -25,12 +27,16 @@ def test_gateway_readiness(gateway):
     assert gateway.is_ready()
 
 
-def test_smoke(route):
+def test_smoke(route, upstream_gateway):
     """
     Tests whether the backend, exposed using the HTTPRoute and Gateway, was exposed correctly,
     having a tls secured endpoint with a hostname managed by MGC
     """
-    backend_client = route.client(verify=False)  # self-signed certificate; TBD
+
+    tls_cert = upstream_gateway.get_tls_cert()
+
+    # assert that tls_cert is used by the server
+    backend_client = HttpxBackoffClient(base_url=f"https://{route.hostnames[0]}", verify=tls_cert)
```

Comment:
would prefer "http or https" logic moved to the upper level, in HTTPRoute class / abstract it to fixture. And just client.get() here

Reply:
Will be part of the next PR with the rework